src/extension/app/store/app.js

```diff
@@ -355,12 +357,25 @@ export class AppStore {
             };
             return environments.some((env) => envChecks[env] && envChecks[env].call(appStore));
           };
-            // assemble plugin config
+          // assemble plugin config
           const plugin = {
             custom: true,
-            id,
+            id: id || `custom-plugin-${i}`,
             condition,
-            button: {
+            pinned,
+            container: containerId,
+          };
+          if (isBadge) {
+            plugin.elements = [{
+              tag: 'sp-badge',
+              text: title,
+              attrs: {
+                variant: badgeVariant?.toLowerCase() || 'default',
+                size: 's',
+              },
+            }];
```

Comment:
I would move this to the rendering phase.

Reply:
You mean not using `plugin.elements` at all, only `plugin.button` with a `isButton` flag?

Reply:
No `plugin.button`, just ` `plugin.isBadge` property.